File: src/main/java/com/koddy/server/coffeechat/application/usecase/query/response/CoffeeChatEachCategoryCounts.java
package com.koddy.server.coffeechat.application.usecase.query.response;

import com.koddy.server.coffeechat.domain.repository.query.response.CoffeeChatCountPerCategory;

public record CoffeeChatEachCategoryCounts(
        long waiting,
        long suggested,
        long scheduled,
        long passed
) {
    public static CoffeeChatEachCategoryCounts from(final CoffeeChatCountPerCategory query) {
        return new CoffeeChatEachCategoryCounts(
                query.waiting(),
                query.suggested(),
                query.scheduled(),
                query.passed()
        );
    }
}

File: src/main/java/com/koddy/server/coffeechat/application/usecase/query/response/CoffeeChatEachCategoryCounts.kt
package com.koddy.server.coffeechat.application.usecase.query.response

import com.koddy.server.coffeechat.domain.repository.query.response.CoffeeChatCountPerCategory

data class CoffeeChatEachCategoryCounts(
    val waiting: Long,
    val suggested: Long,
    val scheduled: Long,
    val passed: Long,
) {
    companion object {
        fun from(query: CoffeeChatCountPerCategory): CoffeeChatEachCategoryCounts =
            CoffeeChatEachCategoryCounts(
                waiting = query.waiting,
                suggested = query.suggested,
                scheduled = query.scheduled,
                passed = query.passed,
            )
    }
}
